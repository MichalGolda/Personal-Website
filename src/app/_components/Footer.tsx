"use client";

import Image from "next/image";
import { useCurrentYear } from "@/app/_hooks/useCurrentYear";
import { Container } from "@/app/_components";

export default function Footer() {
  const currentYear = useCurrentYear();

  return (
    <footer className="py-8">
      <Container className="flex flex-row justify-between items-center">
        <Image src="/logo.svg" width={151} height={29} alt="logo" />
        <span className="text-body text-xs">
          © {currentYear} Wszelkie prawa zastrzeżone.{" "}
          <span className="font-medium">
            Zaprojektowane & Wdrożone przez Michał Gołda
          </span>
        </span>
        <div className="flex flex-row gap-x-4">
          <a href="#">
            <Image src="/github.svg" width={24} height={24} alt="github" />
          </a>
          <a href="#">
            <Image src="/linkedin.svg" width={24} height={24} alt="linkedin" />
          </a>
        </div>
      </Container>
    </footer>
  );
}
