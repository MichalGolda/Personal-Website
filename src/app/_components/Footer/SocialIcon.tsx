import Image from "next/image";

export type FooterIconProps = {
  src: string;
  alt: string;
  href: string;
};

const SocialIcon = ({ src, alt, href }: FooterIconProps) => {
  return (
    <a href={href}>
      <Image src={src} width={24} height={24} alt={alt} />
    </a>
  );
};

export const GithubSocialIcon = () => (
  <SocialIcon
    src="/github.webp"
    alt="github"
    href="https://github.com/michalgolda"
  />
);

export const LinkedInSocialIcon = () => (
  <SocialIcon
    src="/linkedin.webp"
    alt="linkedin"
    href="https://linkedin.com/michalgolda"
  />
);

export default SocialIcon;
