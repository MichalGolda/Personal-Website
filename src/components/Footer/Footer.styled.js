import styled from "styled-components";

export const StyledFooter = styled.footer`
  padding-top: 32px;
  padding-bottom: 32px;
  text-align: center;
  margin-top: 128px;
`;

export const StyledText = styled.p`
  font-weight: 700;
  font-size: 14px;
  color: ${({ theme }) => theme.colors.secondary};
`;
