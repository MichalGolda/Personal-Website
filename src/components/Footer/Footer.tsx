import { StyledFooter, StyledText } from "./Footer.styled";
import { getCurrentYear } from "@/utils/getCurrentYear";

export default function Footer() {
  return (
    <StyledFooter>
      <StyledText>
        © {getCurrentYear()} Michał Gołda. All rights reserved
      </StyledText>
    </StyledFooter>
  );
}
