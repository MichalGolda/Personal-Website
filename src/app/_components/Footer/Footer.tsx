"use client";

import { useCurrentYear } from "@/app/_hooks/useCurrentYear";
import { Container, Logo } from "@/app/_components";
import { GithubSocialIcon, LinkedInSocialIcon } from "./SocialIcon";

export default function Footer() {
  const currentYear = useCurrentYear();

  return (
    <footer className="py-8">
      <Container className="flex flex-col gap-y-4 lg:flex-row justify-between items-center">
        <Logo />
        <span className="text-body text-xs text-center">
          © {currentYear} Wszelkie prawa zastrzeżone. <br />
          <span className="font-medium">
            Zaprojektowane & Wdrożone przez Michał Gołda
          </span>
        </span>
        <div className="flex flex-row gap-x-4">
          <GithubSocialIcon />
          <LinkedInSocialIcon />
        </div>
      </Container>
    </footer>
  );
}
