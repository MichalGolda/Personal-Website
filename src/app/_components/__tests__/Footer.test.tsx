import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import Footer from "../Footer/Footer";

describe("Footer component", () => {
  it("renders without errors", () => {
    render(<Footer />);
  });

  it("renders unchanged", () => {
    const { container } = render(<Footer />);

    expect(container).toMatchSnapshot();
  });
});
