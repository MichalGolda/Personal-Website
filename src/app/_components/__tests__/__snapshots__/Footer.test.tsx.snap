// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component renders unchanged 1`] = `
<div>
  <footer
    class="py-8"
  >
    <div
      class="md:max-w-[1024px] lg:max-w-[1320px] w-full mx-auto md:px-16 px-4 flex flex-col gap-y-4 lg:flex-row justify-between items-center"
    >
      <a
        href="#top"
      >
        <img
          alt="logo"
          data-nimg="1"
          decoding="async"
          height="29"
          loading="lazy"
          src="/logo.svg"
          style="color: transparent;"
          width="151"
        />
      </a>
      <span
        class="text-body text-xs text-center"
      >
        © 
        2024
         Wszelkie prawa zastrzeżone. 
        <br />
        <span
          class="font-medium"
        >
          Zaprojektowane & Wdrożone przez Michał Gołda
        </span>
      </span>
      <div
        class="flex flex-row gap-x-4"
      >
        <a
          href="https://github.com/michalgolda"
        >
          <img
            alt="github"
            data-nimg="1"
            decoding="async"
            height="24"
            loading="lazy"
            src="/github.svg"
            style="color: transparent;"
            width="24"
          />
        </a>
        <a
          href="https://linkedin.com/michalgolda"
        >
          <img
            alt="linkedin"
            data-nimg="1"
            decoding="async"
            height="24"
            loading="lazy"
            src="/linkedin.svg"
            style="color: transparent;"
            width="24"
          />
        </a>
      </div>
    </div>
  </footer>
</div>
`;
